FreeRTOS/FreeRTOS-Kernel
# Style: Make Renesas RXv3 DPFPU port layer consistent with uncrustify.

Description

The following changes are made.

Add '/\* \*INDENT-OFF\* \*/' and '/\* \*INDENT-ON\* \*/' around inline assembly code in C source files.
Add above comments around '#ifdef __cplusplus extern "C" #endif' in portmacro.h.
Add above comments around '#ifdef __cplusplus } #endif' in portmacro.h.
Add '\; /\* \*INDENT-OFF\* \*/' at the top of assembly source file.
Remove '1 tab == 4 spaces!'.
Change tab to spaces.
Beautification by uncrustify.
Beautification by hand (no conflict with uncrustify).
Fix inconsistency which I didn't notice at the commit 386d854: DFPU --> DPFPU.

The following uncrustify and config file are used.

uncrustify: [uncrustify-0.71.0_f-win32](https://sourceforge.net/projects/uncrustify/) 
config: [FreeRTOS/tools/uncrustify.cfg](https://github.com/FreeRTOS/FreeRTOS/commit/4a7a48790d64127f85cc763721b575c51c452833)

Test Steps

There are no changes in the MOT and SREC files (the following total 2 * 3 * 3 = 18 cases) which are built for Renesas RX72N Envision Kit RTOS Demo between before and after.

Total 2 * 3 * 3 = 18 cases:

(1) Define/not define USE_FULL_REGISTER_INITIALISATION in FreeRTOSConfig.h
(2) Define configUSE_TASK_DPFPU_SUPPORT = 1, 2, 0 in FreeRTOSConfig.h
(3) GNURX, CC-RX, ICCRX

Related Issue

Style: Revert uncrustify for portable directories #115
[https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/115#issuecomment-673935703](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/115#issuecomment-673935703)

In addition to a small tip in the above conversation for embedding /\* \*INDENT-OFF\* \*/ into Renesas CC-RX's assembly source file (*.src), I notice that there is another tip to do it. The following source file is fortunately consistent with uncrustify. But I'm not sure whether it may cause some other problems or not.

Especially there are two spaces between ';' and '*', for example:
```
;  * FreeRTOS Kernel V10.3.1
```
instead of one space between ';' and '*', for example:
```
; * FreeRTOS Kernel V10.3.1
```
It may happen that two spaces will be re-formatted to one space by other tool or script.

FreeRTOS/Source/portable/Renesas/RX700v3_DPFPU/port_asm.src

```
; /* *INDENT-OFF* */     <---- HERE
; /*
;  * FreeRTOS Kernel V10.3.1
;  * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
;  *
;  * Permission is hereby granted, free of charge, to any person obtaining a copy of
;  * this software and associated documentation files (the "Software"), to deal in
;  * the Software without restriction, including without limitation the rights to
;  * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
;  * the Software, and to permit persons to whom the Software is furnished to do so,
;  * subject to the following conditions:
;  *
;  * The above copyright notice and this permission notice shall be included in all
;  * copies or substantial portions of the Software.
;  *
;  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
;  * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
;  * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
;  * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
;  * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
;  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
;  *
;  * https://www.FreeRTOS.org
;  * https://github.com/FreeRTOS
;  *
;  */

        .GLB    _vSoftwareInterruptISR
        .GLB    _vSoftwareInterruptEntry

        .SECTION   P,CODE

_vSoftwareInterruptEntry:

    BRA _vSoftwareInterruptISR

        .RVECTOR    27, _vSoftwareInterruptEntry

        .END
```
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
